### DISCUSSION TITLE
Disable popups when launching in Codespaces

### DISCUSSION THREAD

**Discussion by beriberikix** [microsoft/vscode-cpptools#11796] [Q&A]
![image](https://github.com/microsoft/vscode-cpptools/assets/178376/e490f739-262b-4dbc-8939-8998f305ba26)

We're using the extension pack in a devcontainer for use in Codespaces (see [here](https://codespaces.new/golioth/zephyr-training) and [here](https://github.com/golioth/zephyr-training/blob/main/.devcontainer/devcontainer.json).)

We'll be using as part of an online training session and in our testing found those popups distracting to trainees. Is there anyway to disable them from showing up, either via `settings` or some other means?

Thanks!

**Comment by sean-mcmanus** [comment]
@beriberikix I don't believe there is a way. The UI is controlled by VS Code itself (so a request to disable all poups would need to go to them at https://github.com/microsoft/vscode/issues) and extensions control the individual popups.

For the popup from the C/C++ extension shown in the screenshot, it can be disabled by selecting the Yes or No or installing the 1.19.x pre-release version of the extension. But our extension has other popups in other situations and no setting to disable all of them.

It seems to me that it might be good for the training to include how to best handle popups from the tools.

**Reply by beriberikix** [reply]
Thanks for the quick reply @sean-mcmanus! We're actually not interested in disabling _all_ popups, just these specific one on startup. I too find it a bit annoying, even being a regular VS Code / Codespaces user. Dare I say clippy-esque?

Ideally there would be a setting, at least on the C++ and CMake to not show these popups by default, or if they're disable in a `.devcontainer` settings, they are honored. It seems like there's no settings that would facilitate that?

**Reply by bobbrow** [reply]
@sean-mcmanus we should be able to disable the pre-release popup for Codespaces.  I was thinking we used to do that for one of our popups a while ago.  We have an `isCodespaces()` function in common.ts.
